src/Nethermind/Nethermind.State/StateProvider.cs

```diff
@@ -657,9 +658,11 @@ public StateProvider(IScopedTrieStore? trieStore,
             };
         }
 
-        public bool WarmUp(Address address)
+        public bool WarmUp(Address address, out Account? account)
         {
-            return GetState(address) is not null;
+            Debug.Assert(_populatePreBlockCache);
```

Comment:
shouldn't we check that flag for safety?

Reply:
Is only called by prewarmer

Reply:
Can we make it more hidden?